contracts-core/package.json

```diff
@@ -32,9 +33,8 @@
     "ethereum-waffle": "^3.2.0",
     "ethers": "5.7.2",
     "fs-extra": "^9.0.1",
-    "hardhat": "^2.6.8",
+    "hardhat": "^2.17.1",
```

Comment:
The PR description says:

> We're using hardhat version `2.17.0` instead of `2.17.1` to deal possibly monorepo-related error:

But this contradicts that. Curioius which is correct / what the possible issue was?

Reply:
Sorry, so when both `contracts-core` and `umbra-js` hardhat versions are set to `^2.17.1`, we get the following error only in `Umbra-js` when running `yarn test`
```
yarn test
yarn run v1.22.18
$ yarn build && yarn hardhat test
$ tsc --build
$ /Users/user/github/umbra-protocol/node_modules/.bin/hardhat test
Error HH12: Trying to use a non-local installation of Hardhat, which is not supported.
Please install Hardhat locally using npm or Yarn, and try again.

For more info go to https://hardhat.org/HH12 or run Hardhat with --show-stack-traces
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
So one way of solving this was installing a different versions of hardhat in umbra-js, so it would keep its own copy of hardhat.

Reply:
I see that using pinned version `2.17.1` inside `umbra-js` instead `^2.17.1` also works. So I've done that here: https://github.com/ScopeLift/umbra-protocol/pull/577/commits/0a2b406b55ad9dceaa3ca55069e287756b0351c4

Reply:
@garyghayrat i'll be surprised if it works now; perhaps it works on your machine because you already had an umbra-js/node_modules/hardhat around, but I think we can expect that if umbra-js and contracts-core share the same version, we'll see the "non-local hardhat" error. hoping i get proved wrong!

Reply:
¯\_(ツ)_/¯